docs/spec_draft.md

```diff
@@ -254,6 +311,54 @@ the (real, imaginary) pairs.
 // %result: [[1, 2], [3, 4]]
 ```
 
+## stablehlo.negate
+
+`stablehlo.negate(operand) -> result`
+
+### Semantics
+
+Performs element-wise negation of `operand` tensor and produces a `result`
+tensor. For complex types, the operation negates both the real and the imaginary
+parts.
+
+For n-bit signed integer, the negation of $-2^{n-1}$ is implementation
+defined and one of the followings:
+
+  * Saturation to $2^{n-1}-1$
+  * $-2^n-1$
+
+### Operands
+
+| Name | Type |
+|-|-|
+| `operand` | tensor of signed integer, floating-point, or complex element types |
```

Comment:
Unlike the MHLO version, the StableHLO op does not support unsigned types as it does not makes sense. 

The decision is similar to : https://www.tensorflow.org/api_docs/python/tf/math/negative
Moreover,  abs op has related somewhat semantics to negate and the supported types for the mhlo.abs operator has unsigned excluded.

Reply:
Any idea why [shape_inference.cc for kNegate](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/service/shape_inference.cc#L308-L317) supports unsigned types? Should we change it to not support them?

"as it does not makes sense". Does anyone use negate on unsigned types at the moment? E.g. TF? JAX? PyTorch/XLA? If someone does, then we'll need to support that as well for backward compatibility reasons.

Reply:
>> Any idea why supports unsigned types?
Still exploring.

Added the unsigned type. The current behavior is same as that of HLO. 
[Jax](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.neg.html#jax.lax.neg) and XLA support them. From [PyTorch spec](https://pytorch.org/docs/stable/generated/torch.neg.html?highlight=neg#torch.neg), it seems they do also support.

Reply:
"Operands" and "Results" still say "tensor of signed integer...".